### ISSUE TITLE
TorchDialect.cpp, TorchToLinalg.cpp taking a long time to compile

### ISSUE THREAD

**Issue by silvasean** [llvm/torch-mlir#382]
These hurt developer velocity by being on the critical path of edit/compile/run cycles.

TorchToLinalg.cpp could be split along op categories:

- view-like
- elementwise/broadcast_to ops (consider moving BatchNorm here too)
- constructor ops (zeros, ones, ...)
- contracting ops (reduce/conv/pool/matmul/etc.)

For TorchDialect.cpp I'm not sure what is the exact source of the issue (TorchOps.cpp might be taking a long time too). One possible solution could be sharding the ops into multiple cpp files. TF does something using `-gen-op-decls -op-include-regex=REGEX` args to tablegen to split it into multiple .cpp files. See example here in TF (I think they do it for similar reasons): https://github.com/tensorflow/tensorflow/blob/e2d117929e6cc4fa4c7ae3ab965997aecba87bf6/tensorflow/compiler/mlir/tensorflow/BUILD#L196

**Comment by pashu123** [comment]
@silvasean I think this: https://github.com/onnx/onnx-mlir/tree/master/src/Conversion/ONNXToKrnl would be a good reference point for splitting across op categories.

**Comment by silvasean** [comment]
I looked, and -op-include-regex isn't necessarily the best for us right now. See https://bugs.llvm.org/show_bug.cgi?id=52447

**Comment by silvasean** [comment]
We did the split here. Closing.
